streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -226,7 +227,7 @@ public StreamsMetadata getLocalMetadata() {
             if (thisHost.equals(UNKNOWN_HOST)) {
                 return new KeyQueryMetadata(allMetadata.get(0).hostInfo(), Collections.emptySet(), -1);
             }
-            return new KeyQueryMetadata(localMetadata.hostInfo(), Collections.emptySet(), -1);
+            return new KeyQueryMetadata(localMetadata.get().hostInfo(), Collections.emptySet(), -1);
```

Comment:
Will this get us NullPointer? Do we need to get an initial value for localMetadata?

Reply:
Yeah good catch, see above

Reply:
Very good catch. Thanks, @abbccdda .

Reply:
Hmm, why do we still keep it? Based on the reviews for previous version, I believe that there is some strict ordering for getting `localMetadata` initialized to be non-null on L352 first before hitting this logic, but still a null check sound more resilient to me, unless we want to have a NullPointerException to be thrown explicitly.

Reply:
This gets initialized during the rebalance and IQ isn't available until Streams has reached RUNNING.

Reply:
I looked through the StreamsMetadataState and it does seem like it could technically be null if this instance was never assigned any active or standby tasks at all, ever. That really _shouldn't_ happen, but of course it can if you massively over-provisioned your app and we shouldn't throw an NPE over that.

Seems like this is actually an existing bug that we should fix. Then we can improve the initialization check on the side